bp_common/syn/Makefile.verilator

```diff
@@ -84,6 +84,11 @@ $(SIM_DIR)/run_simsc: $(SIM_COLLATERAL)
 	-@grep "PASS" $(SIM_LOG) || echo "FAILED" > $(SIM_ERROR)
 	-@grep "STATS" -A 3 $(SIM_LOG) > $(SIM_REPORT)
 
+sim_dump.sc: VV_OPTS += --trace
```

Comment:
Is this actually needed? The verilation happens in dump.  Or is this for the case where you're sim_dump-ing without having built before?

Reply:
I see. I thought it wasn't working without the target, but maybe I missed something. I don't think the latter would work though, right? You need to build in the dumping to the verilation process, I don't think it can be flipped on after the fact at sim time.

Reply:
I think if you do
```
make clean
make sim_dump.sc
```

It will automatically build a simsc, and if you've set the VV_OPTS in the sim_dump.sc target, it will pass to the dump.sc to build it with dumping.  A cleaner solution may be

```
sim_dump.sc: build_dump.sc
```

Reply:
okay. should I remove this last commit? I think make build_dump.sc sim.sc works as expected, as you said, so sim_dump.sc isn't needed.

Reply:
Can you add this instead of your fix?

sim_dump.sc: build_dump.sc

That'll fix up the dependency at least.  Plus, you need to do build_dump sim_dump for VCS, so might as well have the target in verilator as well, for consistency

Reply:
New issue for this: https://github.com/black-parrot/black-parrot/issues/492

I think we need more than just:
sim_dump.sc: build_dump.sc
to make this work, no? This would build with dump enabled, but wouldn't actually run?

Reply:
oh, you're totally right.

The two options are
```
sim_dump.sc: build_dump.sc
sim_dump.sc:
  $(MAKE) sim.sc

and the way you had it:

sim_dump.sc: VV_OPTS += --trace
...
sim_dump.sc: build.sc
```

I think 2 is better, because it doesn't involve recursive make.